conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/EncodingRegistry.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ObjectArrays;
+import com.google.common.reflect.TypeToken;
+import com.palantir.conjure.java.undertow.lib.Deserializer;
+import com.palantir.conjure.java.undertow.lib.Serializer;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HeaderValues;
+import io.undertow.util.Headers;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/** Orchestrates serialization and deserialization of response and request bodies. */
+final class EncodingRegistry {
```

Comment:
Another design possible instead of `(Encoding default, List<Encoding> encodings)` is `(Encoding default, BiMap<Format, Encoding> formatToEncoding, ContentTypeString -> Format contentTypeToFormat) ` and have the encoding class not have to implement: 
```
    String getContentType();

    boolean supportsContentType(String contentType);
```

Personally, I find the latter easier to follow since it is a bit more explicit that an encoding is tied to a serde mechanism like json or cbor and that the content type mapping is a separate concern.

Reply:
this time on the right thread:

I think we can punt this until later, the `Encoding` interface is public, but only to allow us to test support for other content types outside of this repository. It shouldn't be considered a major break for the runtime module to make changes to Encoding because implementors must understand the risk. I'll add that to the javadoc :-)

Reply:
Fair. On the other hand, I don't think we will want to break later with this since it's a rather minor refactor.